doc/interfaces/denoise.rst

```diff
@@ -0,0 +1,114 @@
+the redundancy of the multi-dimensional information of diffusion-weighted datasets.
+
+The algorithm to perform the PCA-based denoising involves the following steps:
+
+* First, we estimate the local noise variance at each voxel.
+
+* Then, we apply PCA in local patches around each voxel over the gradient directions.
+
+* Finally, we threshold the eigenvalues based on the local estimate of sigma and then do a PCA reconstruction
+
+We are using ``isbi2013_2shell`` dataset. You can also use your own data.
+
+The workflow for the LPCA denoising requires the paths to the diffusion input file,
+as well as the b-values and b-vectors files.
+
+You may want to create an output directory to save the denoised data::
+
+    mkdir denoise_lpca_output
+
+Run the following command::
+
+    dipy_denoise_lpca data/isbi2013/phantom64.nii.gz data/isbi2013/phantom64.bval data/isbi2013/phantom64.bvec --out_dir "denoise_lpca_output"
+
+This command will denoise the diffusion image and save it to the directory
+``denoise_lpca_output``.
+
+In case, you did not specify ``out_dir``, the denoised diffusion image will 
+be saved to a file named ``dwi_lpca.nii.gz`` by default, located in the 
+input directory also by default.
+
+-----------------------------------
+Denoising using Marcenko-Pastur PCA
+-----------------------------------
+
+The PCA-based denoising algorithm exploits the redundancy across the diffusion-
+weighted images. This algorithm has been shown to provide an optimal compromise
+between noise suppression and loss of anatomical information for different 
+techniques such as DTI, spherical deconvolution and DKI.
+
+The basic idea behind the PCA-based denoising algorithms is to remove the components 
+of the data that are classified as noise. The Principal Components classification
+can be performed based on prior noise variance estimates or automatically based 
+on the Marcenko-Pastur distribution. In addition to noise suppression, the PCA
+algorithm can be used to get the standard deviation of the noise.
+
+We will use the ``cfin_multib`` dataset in DIPY to showcase this denoising method.
+As with any other workflow in DIPY, you can also use your own data!
+
+Create a directory where to save the denoised image (e.g.:
+``denoise_mppca_output``)::
+
+    mkdir denoise_mppca_output
+
+In order to run the MPPCA denoising method, we need to specify the location of the 
+diffusion data file, followed by the optional arguments. In this case, we will be 
+specifying the ``patch radius`` value and ``output directory``.
+
+So, we will run the command as::
+
+    dipy_denoise_mppca data/cfin_multib/__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii --patch_radius 10 --out_dir "denoise_mppca_output"
+
+This command will denoise the diffusion image and save it to the specified output directory.
+
+-----------------------
+Denoising using NLMEANS
+-----------------------
+
+Using the non-local means filter ``Coupe08`` and ``Coupe11``, you can denoise 3D or 4D
```

Comment:
I assume ``Coupe08`` and ``Coupe11`` are references to the author's paper, but it is unclear from the text. If the references are to be included, then the citations or papers should be referenced at the bottom.

Reply:
Addressed in 7ba392f